### DISCUSSION TITLE
Interpretable Probabilistic Analysis of AGI Decision Spaces

### DISCUSSION THREAD

**Discussion by SergiusMath** [AGI-course/agi-course.github.io#7] [Ideas]
**Title:** Interpretable Probabilistic Analysis of AGI Decision Spaces
**Type:** Theory + Software

**Overview:**
Consider following example.
AGI system of an arbitrary design (unknown to the observer) makes a set of sequential decisions over a parameterised action space (either discrete or continuous).
We, as the observer, form a high dimensional dataset that describes a mapping from action space to decisions. 
By means of probabilistic modelling, our goal is to restore an interpretable functional space (e.g. surface) that not only corresponds to already observed decsion-making process of AGI, but also provides uncertainty qunatification of its actions and insight into alternative decisions that could have been made.  

**P.S.**
Set of references and pointers can be provided, if required.

**Comment by patham9** [comment]
I needed some time to wrap my head around this. So the idea is for an observer to learn from the interactions of an intelligent agent, to model under what circumstances it is likely to make certain decisions? This can be done indeed. Here is an example to use ONA for a similar task which could be helpful to you:

```
//agent sees a
<a --> [see]>. :|:
//agent moves left
^left. :|:
//agent sees b
<b --> [see]>. :|:
//agent moves right
^right. :|:
//agent sees a
<a --> [see]>. :|:
//what would the agent do to get from seeing b to seeing a?
<(<b --> [see]> &/ ?how) =/> <a --> [see]>>?
//expected: Answer: <(<b --> [see]> &/ ^right) =/> <a --> [see]>>. Truth: frequency=1.000000, confidence=0.241351
```
